### PR TITLE
SLI-368 Migrate components to services to support plugin dynamic loading

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarlint/intellij/SonarLintProjectListener.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.sonarlint.intellij.core.SonarQubeEventNotifications;
+import org.sonarlint.intellij.core.UpdateChecker;
+import org.sonarlint.intellij.editor.CodeAnalyzerRestarter;
+import org.sonarlint.intellij.issue.persistence.LiveIssueCache;
+import org.sonarlint.intellij.trigger.EditorChangeTrigger;
+import org.sonarlint.intellij.ui.SonarLintConsole;
+import org.sonarlint.intellij.util.GlobalLogOutput;
+import org.sonarlint.intellij.util.SonarLintUtils;
+
+public class SonarLintProjectListener implements ProjectManagerListener {
```
[the remark this answers is not in the document]
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarlint/intellij/SonarLintProjectListener.java:36]: A class can't be a service and a listener at the same time. That's why we decided to have a single listener that load services that need some initialization when a project is opened

File: src/main/resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -103,127 +103,14 @@
     </change-notes>
 
     <!-- please see http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html for description -->
-    <idea-version since-build="183.2940.10"/>
+    // we need listeners registration
+    <idea-version since-build="193.4488"/>
```
[the remark this answers is not in the document]
Comment (damien-urruty-sonarsource) [reply src/main/resources/META-INF/plugin.xml:107]: We reverted this change in #244

[the remark this answers is not in the document]
Comment (henryju) [reply src/main/resources/META-INF/plugin.xml:107]: :+1:

File: src/main/java/org/sonarlint/intellij/SonarLintProjectListener.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/SonarLintProjectListener.java:36]: :+1:

**Pull request merged**